enterprise/wan-replication/simulating-max-idle/src/main/java/MultipleMaxIdleSimulator.java

```diff
@@ -0,0 +1,82 @@
+import com.hazelcast.config.Config;
+import com.hazelcast.core.EntryView;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.HazelcastInstanceAware;
+import com.hazelcast.internal.util.Clock;
+import com.hazelcast.map.EntryProcessor;
+import com.hazelcast.map.ExtendedMapEntry;
+import com.hazelcast.map.IMap;
+
+import java.util.Map.Entry;
+import java.util.concurrent.TimeUnit;
+
+class MultipleMaxIdleSimulator extends AbstractMaxIdleSimulator {
+    private static final int MAX_IDLE_MIN_SECONDS = 10;
+    private static final int MAX_IDLE_MAX_SECONDS = 20;
+    private final MaxIdleSimulatingGet<String, String> entryProcessor = new MaxIdleSimulatingGet<>();
+
+    public static void main(String[] args) {
+        AbstractMaxIdleSimulator simulator = new MultipleMaxIdleSimulator();
+        simulator.simulate();
+    }
+
+    private static class MaxIdleSimulatingGet<K, V> implements EntryProcessor<K, V, V>, HazelcastInstanceAware {
+        private transient HazelcastInstance instance;
+        private transient volatile boolean readOnly = true;
+
+        @Override
+        public V process(Entry<K, V> entry) {
+            EntryView<K, V> entryView = instance.<K, V>getMap(MAP_NAME).getEntryView(entry.getKey());
+
+            if (entryView == null) {
+                return null;
+            } else if (Clock.currentTimeMillis() - entryView.getLastUpdateTime() > ttlToMaxIdle((int) entryView.getTtl() / 1000)) {
```

Comment:
Maybe we can note that to have last-update-time, per-entry-stats of map-config should be enabled.

Reply:
Hmm with this PR: https://github.com/hazelcast/hazelcast/pull/23279/files per entry records aren't required.

Reply:
https://github.com/hazelcast/hazelcast-code-samples/pull/560/commits/664e5544ce8a82b46e4878d5ed9c335cc0f2d1c7